sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDbStateStoreProvider.scala

```diff
@@ -0,0 +1,652 @@
+    } catch {
+      case NonFatal(e) => logWarning(s"Error cleaning up files for $this", e)
+    }
+  }
+
+  // Used only for unit tests
+  private[sql] def latestIterator(): Iterator[UnsafeRowPair] = synchronized {
+    val versionsInFiles = fetchFiles(fm, baseDir).map(_.version).toSet
+    var itr = Iterator.empty
+    if (versionsInFiles.nonEmpty) {
+      val maxVersion = versionsInFiles.max
+      if (maxVersion == 0) {
+        return Iterator.empty
+      }
+      // FIXME assuming maxVersion exists in rocksDB
+      val path = getBackupPath(maxVersion)
+      val r: RocksDbInstance = new RocksDbInstance(keySchema, valueSchema, maxVersion.toString)
+      try {
+        r.open(path, rocksDbConf, readOnly = true)
+        return r.iterator(false)
+      } catch {
+        case e: Exception =>
+        // do nothing
+      }
+    }
+    Iterator.empty
+  }
+
+  // making it public for unit tests
+  lazy val rocksDbPath: String = {
+    val checkpointRootLocationPath = new Path(stateStoreId.checkpointRootLocation)
+    val basePath = new Path(
+      localDirectory,
+      new Path(
+        "db",
+        checkpointRootLocationPath.getName + "_" + checkpointRootLocationPath.hashCode()))
+
+    val dir = basePath.toString + Path.SEPARATOR +
+      stateStoreId_.operatorId + Path.SEPARATOR +
+      stateStoreId_.partitionId
+
+    val f: File = new File(dir)
+
+    if (!f.exists()) {
+      logInfo(s"creating rocksDb directory at : $dir")
+      f.mkdirs()
+    }
+    dir
+  }
+
+  private def getBackupPath(version: Long): String = {
+    val checkpointRootLocationPath = new Path(stateStoreId.checkpointRootLocation)
+
+    val basePath = new Path(
+      localDirectory,
+      new Path(
+        "backup",
+        checkpointRootLocationPath.getName + "_" + checkpointRootLocationPath.hashCode()))
+
+    val dir = basePath.toString + Path.SEPARATOR +
+      stateStoreId_.operatorId + Path.SEPARATOR +
+      stateStoreId_.partitionId
+
+    val f: File = new File(dir)
+
+    if (!f.exists()) {
+      logInfo(s"creating rocksDb directory at : $dir")
+      f.mkdirs()
+    }
+
+    dir + Path.SEPARATOR + version
+  }
+
+  private def getTempPath(version: Long): String = {
+    val checkpointRootLocationPath = new Path(stateStoreId.checkpointRootLocation)
+
+    val basePath = new Path(
+      localDirectory,
+      new Path(
+        "tmp",
+        checkpointRootLocationPath.getName + "_" + checkpointRootLocationPath.hashCode()))
+
+    val dir = basePath.toString + Path.SEPARATOR +
+      stateStoreId_.operatorId + Path.SEPARATOR +
+      stateStoreId_.partitionId
+
+    val f: File = new File(dir)
+
+    if (!f.exists()) {
+      logInfo(s"creating rocksDb directory at : $dir")
+      f.mkdirs()
+    }
+
+    dir + Path.SEPARATOR + version
+  }
+
+  // making it public for unit tests
+  def getLocalDirectory: String = localDirectory
+}
+
+object RocksDbStateStoreProvider {
+
+  val ROCKS_DB_BASE_PATH: String = "/media/ephemeral0/spark/rocksdb"
```

Comment:
This should be realistic path, like spark local directory or temporary directory. In many cases it will try to create '/media' in root directory and fail.

Reply:
will make the changes.